CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+cmake_minimum_required( VERSION 3.0 )
+
+project( cubature )
+
+add_library( cubature SHARED 
+    hcubature.c
+    pcubature.c)
+
+add_executable( htest test.c )
+target_link_libraries( htest cubature m )
+
+add_executable( ptest test.c )
+target_link_libraries( ptest cubature m )
+set_target_properties( ptest PROPERTIES COMPILE_FLAGS "-DPCUBATURE" )
```

Comment:
Shouldn't this be https://cmake.org/cmake/help/v3.0/command/target_compile_definitions.html ?